docs/blog/2019-03-25-behind-the-scenes-what-makes-gatsby-great/index.md

```diff
@@ -0,0 +1,496 @@
+
+Gatsby--like other tools--uses the filesystem as a convention for mapping to routes for instance, given the following directory structure:
+
+_also note we expose a [programatic API](/docs/node-apis/#createPages) for dynamic route creation_
+
+```
+├── src/
+  ├── pages/
+    └── about.js
+    └── contact.js
+    └── index.js
+```
+
+The _routes_ (e.g. the URL the user enters or navigates to in the website) `/about`, `/contact`, and `/` will be available in the resulting application. Let's take a look at one of these routes.
+
+```jsx:title=src/pages/contact.js
+import React from "react"
+import { Formik } from "formik"
+import * as yup from "yup"
+
+import { Layout } from "../components/layout"
+
+const handleSubmit = values => {
+  // submit the form
+}
+
+export default function Contact() {
+  return (
+    <Layout>
+      <Formik
+        initialValues={{ email: ``, name: ``, message: `` }}
+        onSubmit={handleSubmit}
+        validationSchema={yup.object().shape({
+          email: yup
+            .string()
+            .email()
+            .required(),
+          name: yup.string().required(),
+          message: yup.string().required(),
+        })}
+      >
+        {props => (
+          <form onSubmit={props.onSubmit}>{/* the rendered form */}</form>
+        )}
+      </Formik>
+    </Layout>
+  )
+}
+```
+
+Pretty vanilla looking component! We are rendering a `form` with some validation and functionality provided by the excellent libraries [`yup`](https://www.npmjs.com/package/yup) and [`Formik`](https://github.com/jaredpalmer/formik). The likelihood that these libraries are used in _all_ routes in our application is unlikely--yet this is traditionally the approach that many take with bundling their client-side JS libraries. This means that even if a particular route (e.g. `/about`) is _not using_ certain libraries that they will still likely be included in a monolithic JavaScript bundle containing all dependencies. However--Gatsby, your friendly _web app compiler_, is a little smarter!
+
+We use code-splitting (enabled via our internalized dependency [Webpack](https://webpackjs.org)), and in particular, our approach prioritizes app-level dependencies (libraries used by the majority or all routes) coupled with route-based code splitting for dependencies that are likely only used on a particular route. To more fully understand this, let's take a look at a sample structure produced by our build process: `gatsby build`.
+
+```title=public/
+├── 404
+│   └── index.html
+├── 9-f5d9c17474395c2890a3.js # highlight-line
+├── about
+│   └── index.html
+├── app-2abedacd9bf03aaca373.js # highlight-line
+├── component---src-pages-404-js-295c3d4f21322761edff.js
+├── component---src-pages-about-js-3997b0d76203b183f5b3.js
+├── component---src-pages-contact-js-34c976efa1482a119a50.js
+├── component---src-pages-index-js-764f0d722c982d3d2789.js
+├── contact
+│   └── index.html # highlight-line
+└── index.html
+```
```

Comment:
Missing here this from `index.html` below:

```
/static/d/686/path---contact-26-a-cd9-NZuapzHg3X9TaN1iIixfv1W23E.json
```

Reply:
It's intentionally simplified--but thank you!

Do you think it's confusing if I omit this? (as I have done here)

Reply:
oh yes - i read the html snipped, remembered there was no json, and then looked for this file and did not find it. to be complete it would be nice to have and we see also where the static files are located.

Reply:
mayby add also an `[... and presized images also]`